backend/tests/data_access/test_repository.py

```diff
@@ -108,6 +110,75 @@ def test_get_artefacts_by_family_name_latest(
     # Assert
     assert len(artefacts) == len(expected_artefacts)
     assert {
-        (artefact.name, artefact.stage.name, artefact.created_at, artefact.version)
+        (
+            artefact.name,
+            artefact.stage.family.name,
+            artefact.stage.name,
+            artefact.created_at,
+            artefact.version,
+        )
         for artefact in artefacts
     } == expected_artefacts
+
+
+def test_get_artefacts_by_family_charm_latest_architectures(
+    db_session: Session, generator: DataGenerator
+):
+    """For latest charms, latest artefacts should include all unique architectures"""
+    # Arrange
+    base = SimpleNamespace(
+        expect=True,
+        name="name-1",
+        version="1",
+        family="charm",
+        track="track-1",
+        stage="edge",
+        day=0,
+        arches=["arch-1"],
+    )
+    specs = [
+        # Base case
+        SimpleNamespace(),
+        # Family is not Charm
+        SimpleNamespace(version="2", family="snap", expect=False),
```

Comment:
Re: [comment](https://github.com/canonical/test_observer/pull/237/files#r1884292510) to make `Artefact` unique constraint `(name, version, family)`

One thing about the `snap_unique` constraint on `(name, version, track)` is that there cannot be a snap or charm with those fields being the same. Probably never going to arise, but just one reason why it'd be nice to figure out how to include the family on uniqueness.

Reply:
It's an unlikely issue, but could arise if there is a snap and charm with the same name (and version and track).

Reply:
Good point. We could add a foreign key on Artefact for family_id as you mentioned. Feel free to do it in this PR or some later PR